### ISSUE TITLE
multiline-ternary and operator-linebreak rules don't work well with JSX

### ISSUE THREAD

**Issue by Linksku** [eslint/eslint#9916]
* **ESLint Version:**
4.16.0

* **Node Version:**
9.3.0

* **npm Version:**
5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

{
  "parser": "babel-eslint",
  "parserOptions": {
    "ecmaVersion": 2017,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "extends": ["airbnb", "prettier"],
  "plugins": ["jsx-a11y", "prettier", "react"],
  "env": {
    "browser": true,
    "commonjs": 1,
    "es6": true,
    "jest": true
  },
  "rules": {
    "prettier/prettier": [
      "warn",
      {
        "singleQuote": true,
        "trailingComma": "all",
        "bracketSpacing": true,
        "jsxBracketSameLine": false,
        "arrowParens": "always"
      }
    ],
    "react/react-in-jsx-scope": 0,
    "prefer-const": 1,
    "react/prop-types": 1,
    "react/jsx-filename-extension": [2, { "extensions": [".js", ".jsx"] }],
    "import/no-extraneous-dependencies": 0,
    "import/no-unresolved": 0,
    "import/extensions": 0,
    "no-underscore-dangle": ["warn", { "allowAfterThis": true }],
    "react/jsx-tag-spacing": 1,
    "react/jsx-closing-bracket-location": [2, "tag-aligned"],
    "no-unused-expressions": 1, /* temporary make into warning until do expressions are supported */
    "no-param-reassign": 1,
    "react/prefer-stateless-function": 1,
    "prefer-destructuring": 1,
    "no-plusplus": 1,
    "arrow-body-style": [2, "as-needed"],
    "operator-linebreak": [2, "before", { "overrides": { "=": "after" } }],
    "multiline-ternary": [2, "always-multiline"],
    "no-unneeded-ternary": 2,
    "react/jsx-no-bind": 1,
    "no-alert": 2,
    "class-methods-use-this": 1,
    "react/jsx-wrap-multilines": [2, {
       "condition": "parens-new-line"
    }],
    "newline-per-chained-call": [2, { "ignoreChainWithDepth": 3 }]
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

<div>
  {cond ? (
      <br />
    ) : null}
</div>

```

```bash
node_modules/.bin/eslint client --ext=js --quiet
```

**What did you expect to happen?**
I expected it to be an error. I expected this to be the only valid code:
```js

<div>
  {cond
    ? (
      <br />
    )
    : null}
</div>

```

**What actually happened? Please include the actual, raw output from ESLint.**
It passed the linter.
